File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/enumeration/Approver.java
package org.babyfish.jimmer.sql.model.enumeration;

import org.babyfish.jimmer.sql.DatabaseValidationIgnore;
import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.GeneratedValue;
import org.babyfish.jimmer.sql.Id;
import org.jetbrains.annotations.Nullable;

@DatabaseValidationIgnore
@Entity
public interface Approver {

    @Id
    @GeneratedValue
    long id();

    String name();

    @Nullable
    Gender gender();
}

File: project/jimmer-sql-kotlin-test-model/src/main/kotlin/org/babyfish/jimmer/sql/kt/model/enumeration/Approver.kt
package org.babyfish.jimmer.sql.kt.model.enumeration

import org.babyfish.jimmer.sql.DatabaseValidationIgnore
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Id

@DatabaseValidationIgnore
@Entity
interface Approver {

    @Id
    val id: Long

    val name: String

    val gender: Gender?
}